examples/keras_recipes/creating_tfrecords.py

```diff
@@ -0,0 +1,290 @@
+bbox: [x,y,width,height], object bounding box coordinates
+area: float, area of the bounding box
+iscrowd: 0 or 1, is single object or a collection
+```
+
+### Looking at the annotations of a single data sample
+"""
+
+pprint.pprint(annotations[60])
+
+"""
+## Parameters
+
+`n_samples` is the number of data samples on each TFRecord file.
+
+`n_tfrecods` is total number of TFRecords that we will create.
+"""
+
+n_samples = 4096
+n_tfrecods = len(annotations) // n_samples
+if len(annotations) % n_samples:
+    n_tfrecods += 1  # add one record if there are any remaining samples
+
+if not os.path.exists(tfrecords_dir):
+    os.makedirs(tfrecords_dir)  # creating TFRecords output folder
+
+"""
+## TFRecords functions
+"""
+
+
+def image_feature(value):
+    """Returns a bytes_list from a string / byte."""
+    return tf.train.Feature(
+        bytes_list=tf.train.BytesList(value=[tf.io.encode_jpeg(value).numpy()])
+    )
+
+
+def bytes_feature(value):
+    """Returns a bytes_list from a string / byte."""
+    return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value.encode()]))
+
+
+def float_feature(value):
+    """Returns a float_list from a float / double."""
+    return tf.train.Feature(float_list=tf.train.FloatList(value=[value]))
+
+
+def int64_feature(value):
+    """Returns an int64_list from a bool / enum / int / uint."""
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+
+def float_feature_list(value):
+    """Returns a list of float_list from a float / double."""
+    return tf.train.Feature(float_list=tf.train.FloatList(value=value))
+
+
+def create_example(image, path, example):
+    feature = {
+        "image": image_feature(image),
+        "path": bytes_feature(path),
+        "area": float_feature(example["area"]),
+        "bbox": float_feature_list(example["bbox"]),
+        "category_id": int64_feature(example["category_id"]),
+        "id": int64_feature(example["id"]),
+        "image_id": int64_feature(example["image_id"]),
+    }
+    return tf.train.Example(features=tf.train.Features(feature=feature))
+
+
+def parse_tfrecord_fn(example):
+    feature_description = {
+        "image": tf.io.FixedLenFeature([], tf.string),
+        "path": tf.io.FixedLenFeature([], tf.string),
+        "area": tf.io.FixedLenFeature([], tf.float32),
+        "bbox": tf.io.VarLenFeature(tf.float32),
+        "category_id": tf.io.FixedLenFeature([], tf.int64),
+        "id": tf.io.FixedLenFeature([], tf.int64),
+        "image_id": tf.io.FixedLenFeature([], tf.int64),
+    }
+    example = tf.io.parse_single_example(example, feature_description)
+    example["image"] = tf.io.decode_jpeg(example["image"], channels=3)
+    example["bbox"] = tf.sparse.to_dense(example["bbox"])
+    return example
+
+
+"""
+## Generating TFRecords
+
+The generated TFRecrods will have the file name format `file_{number}.tfrec`. This is
+optional, but writing the number of files at the file name can make counting easier.
```

Comment:
Suggestion:

- Minor refactoring
- Fixing misspelling ("TFRecord")

```suggestion
Let's generate the COCO2017 data in the TFRecord format. The format will be
`file_{number}.tfrec` (this is optional, but including the number sequences in the file
names can make counting easier).
```